internal/libs/autofile/autofile.go

```diff
@@ -103,7 +108,11 @@ func (af *AutoFile) closeFileRoutine() {
 	for {
 		select {
 		case <-af.closeTicker.C:
-			_ = af.closeFile()
+			if atomic.LoadUint64(&af.nWrites) != 0 {
+				// Only invoke .closeFile if we had some writes.
+				_ = af.closeFile()
```

Comment:
Why doesn't this result in the file being open forever?

Seems like we should avoid opening the logger file if we did 0 writes?

Reply:
e.g. only generate the file ID & open it when its first written to

Reply:
> Why doesn't this result in the file being open forever?

In the very least it keeps the file just as open as we previously did, and here it is once instead of opening and closing every second. When we terminate with a signal such as Ctrl+C, we invoke closeFile, so nope it doesn't open it forever.